atlasdb-commons/src/main/java/com/palantir/atlasdb/futures/AtlasFutures.java

```diff
@@ -93,6 +93,10 @@ public void close() {
             return listenableFuture.get();
         } catch (ExecutionException e) {
             throw Throwables.rewrapAndThrowUncheckedException(e.getCause());
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
I think that Throwables.rewrapAndThrow handles interrupted exception?

Reply:
Calls down to:

```java
    private static RuntimeException createPalantirRuntimeException(String newMessage, Throwable ex) {
        if (ex instanceof InterruptedException || ex instanceof InterruptedIOException) {
            Thread.currentThread().interrupt();
            return new PalantirInterruptedException(newMessage, ex);
        }
        return new PalantirRuntimeException(newMessage, ex);
    }
```

Reply:
Good call, that's deep in there.

Reply:
yeah, it wasn't my decision to make the PalantirInterruptedException, but I do know it exists :)